rekcurd/data_servers/__init__.py

```diff
@@ -0,0 +1,133 @@
+        return str(local_filepath)
+
+    def upload_model(self, request_iterator: Iterator[rekcurd_pb2.UploadModelRequest]) -> str:
+        first_req = next(request_iterator)
+        filepath = first_req.path
+        valid_path = convert_to_valid_path(filepath)
+        if filepath != str(valid_path):
+            raise Exception(f'Error: Invalid file path specified -> {filepath}')
+
+        local_filepath = Path(self._api_handler.LOCAL_MODEL_DIR, valid_path.name)
+        local_filepath.parent.mkdir(parents=True, exist_ok=True)
+        with local_filepath.open(mode='wb') as f:
+            f.write(first_req.data)
+            for request in request_iterator:
+                f.write(request.data)
+            del first_req
+            f.close()
+        self._api_handler.upload(filepath, str(local_filepath))
+        return str(local_filepath)
+
+    def get_evaluation_data_path(self, filepath: str) -> str:
+        valid_path = convert_to_valid_path(filepath)
+        if filepath != str(valid_path):
+            raise Exception(f'Error: Invalid file path specified -> {filepath}')
+
+        local_filepath = Path(self._api_handler.LOCAL_EVAL_DIR, valid_path.name)
+        if not local_filepath.exists():
+            local_filepath.parent.mkdir(parents=True, exist_ok=True)
+            self._api_handler.download(filepath, str(local_filepath))
+        return str(local_filepath)
+
+    def _get_eval_result(self, filepath: str, suffix: str):
+        valid_path = convert_to_valid_path(filepath)
+        if filepath != str(valid_path):
+            raise Exception(f'Error: Invalid file path specified -> {filepath}')
+
+        local_filepath = Path(self._api_handler.LOCAL_EVAL_DIR, valid_path.name+suffix)
+        if not local_filepath.exists():
+            local_filepath.parent.mkdir(parents=True, exist_ok=True)
+            self._api_handler.download(filepath+suffix, str(local_filepath))
+        return str(local_filepath)
+
+    def get_eval_result_summary(self, filepath: str) -> str:
+        return self._get_eval_result(filepath, self.__EVALUATE_RESULT)
+
+    def get_eval_result_detail(self, filepath: str) -> str:
+        return self._get_eval_result(filepath, self.__EVALUATE_DETAIL)
+
+    def upload_evaluation_data(self, request_iterator: Iterator[rekcurd_pb2.UploadEvaluationDataRequest]) -> str:
+        first_req = next(request_iterator)
+        filepath = first_req.data_path
+        valid_path = convert_to_valid_path(filepath)
+        if filepath != str(valid_path):
+            raise Exception(f'Error: Invalid evaluation file path specified -> {filepath}')
+
+        local_filepath = Path(self._api_handler.LOCAL_EVAL_DIR, valid_path.name)
+        local_filepath.parent.mkdir(parents=True, exist_ok=True)
+        with local_filepath.open(mode='wb') as f:
+            f.write(first_req.data)
+            for request in request_iterator:
+                f.write(request.data)
+            del first_req
+            f.close()
```

Comment:
Do we need `f.close` even though we use `with` statement?

Reply:
I forgot the case but I got the error without using `f.close` in `with` statement when I boot an application.

Reply:
No. It's my misunderstanding. I will remove them. Thank you.
https://github.com/rekcurd/rekcurd-python/pull/23/files